assets/js/onboard-tasks.js

```diff
@@ -0,0 +1,146 @@
+/* global prplDocumentReady */
+/*
+ * Onboard tasks
+ *
+ * A script to process the onboarding task checklist.
+ *
+ * Dependencies: progress-planner-document-ready
+ */
+
+/**
+ * Process the onboarding task checklist.
+ */
+// eslint-disable-next-line no-unused-vars
+async function progressPlannerOnboardTasks() {
+	const tasksElement = document.getElementById( 'prpl-onboarding-tasks' );
+	const timeToWait = 2000;
+
+	if ( ! tasksElement ) {
+		return;
+	}
+
+	// Display the tasks.
+	tasksElement.style.display = 'block';
+
+	const listItems = tasksElement.querySelectorAll( 'li' );
+
+	// Create an array of Promises
+	const tasks = Array.from( listItems ).map( ( li, index ) => {
+		return new Promise( ( resolve ) => {
+			li.classList.add( 'prpl-onboarding-task-loading' );
+
+			setTimeout(
+				() => {
+					const taskCompleted =
+						'true' === li.dataset.prplTaskCompleted;
+					const classToAdd = taskCompleted
+						? 'prpl-onboarding-task-completed'
+						: 'prpl-onboarding-task-not-completed';
+					li.classList.remove( 'prpl-onboarding-task-loading' );
+					li.classList.add( classToAdd );
+
+					// Update total points.
+					if ( taskCompleted ) {
+						const totalPointsElement = document.querySelector(
+							'#prpl-onboarding-tasks .prpl-onboarding-tasks-total-points'
+						);
+						const totalPoints = parseInt(
+							totalPointsElement.textContent
+						);
+						const taskPoints = parseInt(
+							li.querySelector( '.prpl-suggested-task-points' )
+								.textContent
+						);
+						totalPointsElement.textContent =
+							totalPoints + taskPoints + 'pt';
+					}
+
+					resolve(); // Mark this task as complete.
+				},
+				( index + 1 ) * timeToWait
+			);
+		} );
+	} );
+
+	// Wait for all tasks to complete.
+	await Promise.all( tasks );
+
+	// We add a small delay to make sure the user sees if the last task is completed and total points.
+	await new Promise( ( resolve ) => setTimeout( resolve, timeToWait ) );
+
+	// Set the data-onboarding-finished attribute.
+	tasksElement.setAttribute( 'data-onboarding-finished', 'true' );
+
+	// Redirect if scanning is finished.
+	onBoardRedirect( 'onboardTasks' );
+}
+
+/**
+ * Redirect user to the stats page after onboarding or plugin upgrade.
+ * On onboard screen we redirect only if both post scanning and onboarding tasks are finished.
+ *
+ * @param {string} context The context of the redirect.
+ */
+function onBoardRedirect( context = '' ) {
```

Comment:
We should rename to `prplOnboardRedirect` to avoid conflicts, and for consistency we could do `const prplOnboardRedirect = ( context = '' ) => {.....`